sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ConditionalExpressionSuite.scala

```diff
@@ -113,6 +113,35 @@ class ConditionalExpressionSuite extends SparkFunSuite with ExpressionEvalHelper
     assert(CaseWhen(Seq((c2, c4_notNull), (c3, c5))).nullable === true)
   }
 
+  test("case when - nullability of nested types") {
```

Comment:
Can you check both codegen/non-codegen paths by using `checkEvaluation`?

Reply:
Ok, no problem. Will do.